lib/IRGen/IRGen.cpp

```diff
@@ -807,6 +808,7 @@ static void initLLVMModule(const IRGenModule &IGM, ModuleDecl &M) {
 std::pair<IRGenerator *, IRGenModule *>
 swift::irgen::createIRGenModule(SILModule *SILMod, StringRef OutputFilename,
                                 StringRef MainInputFilenameForDebugInfo,
+                                StringRef PrivateDiscriminator,
```

Comment:
BTW, if you like to sprinkle `consts` as far as I know you can say `const StringRef foo` to declare that there is no assignment to `foo` later on.

Reply:
It's a good nit to pick at some point, but I worry most about ownership semantics for StringRef over the potential for hidden re-assignments.  Even the `const`-applications here are more semantic signals that e.g. a given frontend action doesn't emit diagnostics instead of a syntactic assignment block, which is kind of cool.

Reply:
Don't quite follow the second sentence, but yes, the ownership is more problematic in general than the reassignments. In this case the function is short enough that searching for reassignments is not that bad, so it really is a nit.